manticore/platforms/evm_world_state.py

```diff
@@ -0,0 +1,468 @@
+import logging
+from abc import ABC, abstractmethod
+from eth_typing import ChecksumAddress, URI
+from io import TextIOBase
+from typing import Dict, List, Optional, Set, Tuple, Union
+from urllib.parse import ParseResult, urlparse
+from web3 import Web3
+from ..core.smtlib import Array, ArrayVariable, BitVec, BitVecConstant, BitVecITE, BitVecZeroExtend, ConstraintSet
+from ..ethereum.state import State
+from ..exceptions import EthereumError
+
+logger = logging.getLogger(__name__)
+
+
+# sam.moelius: map records which (symbolic) offsets have been written to.  data holds the values
+# written.
+
+
+class Storage:
+    def __init__(self, constraints: ConstraintSet, address: int):
+        """
+        :param constraints: the ConstraintSet with which this Storage object is associated
+        :param address: the address that owns this storage
+        """
+        self._data = constraints.new_array(
+            index_bits=256,
+            value_bits=256,
+            name=f"STORAGE_DATA_{address:x}",
+            avoid_collisions=True,
+            # sam.moelius: The use of default here creates unnecessary if-then-elses.  See
+            # ArrayProxy.get in expression.py.
+            # default=0,
+        )
+
+    def __getitem__(self, offset: Union[int, BitVec]) -> Union[int, BitVec]:
+        return self.get(offset, 0)
+
+    def get(self, offset: Union[int, BitVec], default: Union[int, BitVec]) -> Union[int, BitVec]:
+        if not isinstance(default, BitVec):
+            default = BitVecConstant(256, default)
+        return self._data.get(offset, default)
+
+    def set(self, offset: Union[int, BitVec], value: Union[int, BitVec]):
+        self._data[offset] = value
+
+    def get_items(self) -> List[Tuple[Union[int, BitVec], Union[int, BitVec]]]:
```

Comment:
Looks like an Array method? 
_Need to check where it is used_

Reply:
> Looks like an Array method?

Sorry, I don't understand the question.  What is mean by "an Array method"?

> _Need to check where it is used_

It is currently used only in `EVMWorld.get_storage_items` (which I added back in abe52bc).

Reply:
I mean that it looks like it would be better suited as an member method of the type Array.
It does not use anything of Storage object and it pips into Array's secret-ish elements.

Maybe we should consider having this method as a member method of Array. (If that's not already there?)